submission/persistence/src/main/kotlin/ac/uk/ebi/biostd/persistence/mapping/SubmissionMapper.kt

```diff
@@ -75,32 +75,39 @@ class SubmissionMapper(
 }
 
 private object SectionMapper {
-    fun toSection(section: ExtendedSection, index: Int) = SectionDb(section.accNo, section.type).apply {
-        order = index
-        attributes = toAttributes(section.attributes).mapTo(sortedSetOf(), ::SectionAttribute)
-        links = section.links.mapIndexed(::toLinks).flatten().toSortedSet()
-        files = section.files.mapIndexed(::toFiles).flatten().toSortedSet()
-        sections = section.extendedSections.mapIndexed(::toSections).flatten().toSortedSet()
-
-        section.libraryFile?.let { libFile ->
-            libraryFile = LibraryFileDb(libFile.name).apply {
-                files = libFile.referencedFiles.map { EntityMapper.toRefFile(it) }.toSet()
+    fun toSection(section: ExtendedSection, parentSubmission: SubmissionDb?, index: Int) =
+        SectionDb(section.accNo, section.type).apply {
+            order = index
+            attributes = toAttributes(section.attributes).mapTo(sortedSetOf(), ::SectionAttribute)
+            links = section.links.mapIndexed(::toLinks).flatten().toSortedSet()
+            files = section.files.mapIndexed(::toFiles).flatten().toSortedSet()
+            submission = parentSubmission
+            sections = section.extendedSections
+                .mapIndexed { index, section -> toSections(index, section, parentSubmission) }.flatten().toSortedSet()
+
+            section.libraryFile?.let { libFile ->
+                libraryFile = LibraryFileDb(libFile.name).apply {
+                    files = libFile.referencedFiles.map { EntityMapper.toRefFile(it, libFile.name) }.toSet()
```

Comment:
this lambda looks really verbose maybe is a good time to make a private helper method.